runtime/commonMain/src/kotlinx/serialization/SerialDescriptorBuilder.kt

```diff
@@ -200,7 +200,35 @@ public fun SerialDescriptorBuilder.setDescriptor(typeDescriptor: SerialDescripto
  */
 @ImplicitReflectionSerializer
 public inline fun <reified T> SerialDescriptorBuilder.setDescriptor(): SerialDescriptor {
-    return HashSetClassDesc(serializer<T>().descriptor)
+    return setDescriptor(serializer<T>().descriptor)
+}
+
+/**
+ * Creates a [List] out of a child descriptors retrieved via [SerialDescriptor.getElementDescriptor].
+ *
+ * Size of a list is equal to [SerialDescriptor.elementsCount].
+ */
+public fun SerialDescriptor.elementDescriptors(): List<SerialDescriptor> {
```

Comment:
Not sure why these functions should belong to this file

Reply:
I'd rather deprecate it without replacement.
I was just tempted to remove `Shorthands.kt` at all, this file was the most relevant one

Reply:
But they are literally shorthands. And file itself was not doing any harm at all

Reply:
I'd like to avoid "util" files in the core package similarly to how Java doesn't have `FooUtil` in the JDK.
Such files don't help with API exploration and just look like a pile of unrelated methods

Reply:
In contrary to Java, file names are not exposed in Kotlin neither used as namespaces.

Reply:
I mean, the problem is with the semantics of that, not with the filename.
Java doesn't have "semantically *Util" as well